frame/composable-support/src/abstractions/nonce/mod.rs

```diff
@@ -0,0 +1,149 @@
+use sp_std::marker::PhantomData;
+
+use crate::abstractions::utils::{increment::Increment, start_at::StartAtValue};
+
+use codec::FullCodec;
+use frame_support::{
+	pallet_prelude::StorageValue,
+	traits::{Get, StorageInstance},
+};
+
+#[allow(clippy::disallowed_types)]
+use frame_support::pallet_prelude::ValueQuery;
+
+#[cfg(test)]
+mod test_storage_nonce;
+
+/// An extension trait for [`StorageValue`]s that are used as a [nonce](nonce).
+///
+/// [nonce]: <https://www.investopedia.com/terms/n/nonce.asp>
+pub trait StorageNonce<T, S, I>: 'static
```

Comment:
I'd rename the trait to `Nonce`, as technically the trait does not assume that it is operating on storage.